CONTRIBUTING.md

```diff
@@ -101,9 +91,36 @@ source activate pytorch-myfeature
 python setup.py build develop
 ```
 
-#### C++ Development tips
+## C++ Development tips
+
+If you are working on the C++ code, there are a few important things that you
+will want to keep in mind:
+
+1. How to rebuild only the code you are working on, and
+2. How to make rebuilds in the absence of changes go faster.
+
+### Build only what you need.
 
-When you are developing on the C++ side of things, the environment variables `DEBUG` and `NO_CUDA` are helpful.
+`python setup.py build` will build everything, but since our build system is
+not very optimized for incremental rebuilds, this will actually be very slow.
+Far better is to only request rebuilds of the parts of the project you are
+working on:
+
+- Working on `torch/csrc`?  Run `python setup.py build_ext` to rebuild.
```

Comment:
I don't think `build_ext` is enough if you're in develop mode. It's safer to do `python setup.py develop`